posthog/models/event/sql.py

```diff
@@ -185,6 +185,75 @@
     )
 )
 
+
+KAFKA_EVENTS_RECENT_TABLE_JSON_SQL = lambda: (
+    EVENTS_TABLE_BASE_SQL
+    + """
+    SETTINGS kafka_skip_broken_messages = 100
+"""
+).format(
+    table_name="kafka_events_recent_json",
+    cluster=settings.CLICKHOUSE_CLUSTER,
+    engine=kafka_engine(topic=KAFKA_EVENTS_JSON, group="group1_recent"),
+    extra_fields="",
+    materialized_columns="",
+    indexes="",
+)
+
+EVENTS_RECENT_TABLE_JSON_MV_SQL = (
+    lambda: """
+CREATE MATERIALIZED VIEW IF NOT EXISTS events_recent_json_mv ON CLUSTER '{cluster}'
+TO {database}.{target_table}
+AS SELECT
+uuid,
+event,
+properties,
+timestamp,
+team_id,
+distinct_id,
+elements_chain,
+created_at,
+person_id,
+person_created_at,
+person_properties,
+group0_properties,
+group1_properties,
+group2_properties,
+group3_properties,
+group4_properties,
+group0_created_at,
+group1_created_at,
+group2_created_at,
+group3_created_at,
+group4_created_at,
+person_mode,
+_timestamp,
+_offset
+FROM {database}.kafka_events_json
+""".format(
+        target_table=EVENTS_RECENT_DATA_TABLE(),
+        cluster=settings.CLICKHOUSE_CLUSTER,
+        database=settings.CLICKHOUSE_DATABASE,
+    )
+)
+
+EVENTS_RECENT_TABLE_SQL = lambda: (
+    EVENTS_TABLE_BASE_SQL
+    + """PARTITION BY toYYYYMMDD(_timestamp)
```

Comment:
I'd probably partition this by maybe `toStartOfHour(_timestamp)` because that would be roughly 168 partitions (not too many) and we will TTL out data more cleanly and not so much in big daily blocks, just hourly blocks.

24 hours in a day * 7 day retention = 168 partitions for this table

168 partitions is well under the general 1k max rule of thumb